hatchery/src/world_snapshot.rs

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::collections::BTreeMap;
+
+use dallo::ModuleId;
+
+use crate::error::Error;
+use crate::snapshot::{MemoryPath, SnapshotId};
+use crate::world::World;
+use rkyv::{Archive, Deserialize, Serialize};
+
+pub const WORLD_SNAPSHOT_ID_BYTES: usize = 32;
+#[derive(
+    Debug,
+    PartialEq,
+    Eq,
+    Archive,
+    Serialize,
+    Deserialize,
+    PartialOrd,
+    Ord,
+    Hash,
+    Clone,
+    Copy,
+)]
+pub struct WorldSnapshotId([u8; WORLD_SNAPSHOT_ID_BYTES]);
```

Comment:
This should have documentation since it is exported

Reply:
ok